hedera-node/src/main/java/com/hedera/services/records/TransactionRecordService.java

```diff
@@ -66,6 +67,14 @@ public void includeChangesToToken(Token token) {
 			}
 			txnCtx.setCreated(serialNumbers);
 		}
+
+		if (token.hasBurnedUniqueTokens()) {
```

Comment:
When burning we should not add `serialNumbers` to the transaction record. They are added only on creation